components/atom/switch/src/index.js

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import AtomLabel from '../../label/src/index'
+
+const BASE_CLASS = 'sui-AtomSwitch'
+
+const SIZES = {
+  MEDIUM: 'medium',
+  LARGE: 'large'
+}
+
+const TYPES = {
+  TOGGLE: 'toggle',
+  SELECT: 'select',
+  SINGLE: 'single'
+}
+
+class AtomSwitch extends Component {
+  constructor (props) {
+    super(props)
+
+    const {size, disabled, labelLeft, labelRight, type, name, label, labelOptionalText, onToggle} = props
+    this.onToggle = onToggle
+    this.size = size
+    this.type = type
+    this.state = {
+      toggle: false,
+      disabled: disabled,
+      isFocus: false,
+      labelLeft: labelLeft,
+      labelRight: labelRight,
+      name: name,
+      label: label,
+      labelOptionalText: labelOptionalText
+    }
+
+    window.addEventListener('keydown', this.keyBindings.bind(this), true)
+  }
+
+  keyBindings = (event) => {
+    if (!this.state.isFocus) {
+      return
+    }
+
+    if (event.defaultPrevented) {
+      return
+    }
+
+    switch (event.key) {
+      case 'Enter':
+        this.toggleSwitch()
+        event.preventDefault()
+        break
+      case ' ':
+        this.toggleSwitch()
+        event.preventDefault()
+        break
+    }
+  }
+
+  toggleSwitch = function () {
+    if (this.state.disabled) {
+      return
+    }
+    if (this.onToggle) {
+      this.onToggle(!this.state.toggle)
+    }
+    this.setState({toggle: !this.state.toggle})
+  }.bind(this)
+
+  activateToggle = function () {
+    if (this.state.disabled) {
+      return
+    }
+    this.setState({toggle: true})
+  }.bind(this)
+
+  deactivateToggle = function () {
+    if (this.state.disabled) {
+      return
+    }
+    this.setState({toggle: false})
+  }.bind(this)
+
+  focusSwitch = function () {
+    this.setState({isFocus: true})
+  }.bind(this)
+
+  focusOutSwitch = function () {
+    this.setState({isFocus: false})
+  }.bind(this)
+
+  render () {
+    let divClass = (this.state.toggle || this.type === TYPES.SELECT) ? cx(BASE_CLASS, 'active') : cx(BASE_CLASS)
```

Comment:
You really can reduce that code and make it more readable and avoid unnecessary executions.
Check classNames docs